cg/services/orders/order_service/order_service.py

```diff
@@ -25,7 +26,8 @@ def get_order(self, order_id: int) -> OrderResponse:
         return create_order_response(order=order, summary=summary)
 
     def get_orders(self, orders_request: OrdersRequest) -> OrdersResponse:
-        orders, total_count = self.store.get_orders(orders_request)
+        order_query_params: OrderQueryParams = self._get_order_query_params(orders_request)
```

Comment:
If we are making this change, we should probably do it in the OrdersRequest already? I do not see the gain in reading the args into a pydantic object to cast into a slightly different one moments after?

Reply:
I added a diagram showing the communication between the layers. 

Passing the OrderRequest, dependent on requirements from the outside, all the way to the CRUD you are forced to change logic in all layers where the OrderRequest interacts with something. I do not think the OrderRequest achieves the decoupling, it rather couples multiple layers.

Having the Request translated into a different model that is passed on deeper into the system keeps the request logic isolated in the OrderService.

The tradeoff: More complexity (to what degree is arguable) for better separation of concerns and more flexibility of the modules.

Reply:
My arguments are the following:

This service is simple and the only thing it does is get a request from TB which basically consists of all the filtering parameters in the backend. While I do agree that the request from TB in principle is different from the database filtering options, _today_ the TB request is basically the database parameters . Adding extra complexities for a hypothetical flexibility which there currently is no need for is something I would prefer we wait with until we have a concrete need for it. To me, there are no immediate benefits and this separation of concerns is better acted upon when there is an actual difference. 

In short: I do not see any gain from the added flexibility and I have heard of no plans requiring such a change. I will gladly put this decoupling in place when there is a clear need for it, but to me it just adds confusion currently.

Reply:
This PR, if not done in this way will result in what I explained.
We would need to modify the OrderRequest to allow for the behaviour and it would trickle down. I'd argue this exact situation already shows the need.

Reply:
Summarizing a discussion from Slack:

IO: Technically yes - or at least it shows that there is a difference between what is in the request and how the filtering is done. Does it warrant a new object to me? No. If I wrote the PR I would probably add `workflows: Annotated[list[Workflow], BeforeValidator(set_workflows)] = Field(alias="workflow")` to the OrderRequest and be done with that. The difference between the FE workflow parameter and the filtering done on workflows is "close enough" in my mind.

To me a clearer need for the separation would be if we in the FE request suddenly had some parameters which would go towards Housekeeper and some parameters which would go to statusdb. At that point it would make no sense to send the OrderRequest object to both the store and to the housekeeper api for them to only use a fraction of the parameters in the passed object.

CO: To me, the OrderRequest violates SRP.
I also just like that the flow is documented more explictly:
you get a request, that requires a query, and you translate that request to params for the store
Thematically the "Request" and "Response" syntax is more something of the Front end layer to me

IO: And to me we are adding unneeded complexity if we add a marginally different object to remedy that
To me it is mostly flavour, though. I can't argue against it (CO's design) being more flexible and robust

CO: Yes, I think neither is wrong. And we just agree to disagree

At this point, unless anyone else has strong opinions, I would leave it up to CO to decide. I have made my points, he has made his. I think there are pros and cons with both options, and the choice will come down to what weight you give them.

Reply:
My opinion: If this is a hotfix it should not include any refactoring. Bug in prod = fix ASAP, make a bare minimum patch that does not introduce any bugs and merge. Here there is some refactoring that is causing debate and delaying the bug patch. The changes introduced here should have been discussed in a refinement session. Having said that, it seems this solution solves the issue, so I will approve it, although I agree with @islean that a much simpler solution would have sufficed